### PR TITLE
add meetingStartDurationMs event

### PR DESCRIPTION
**Pull request by dingyishen-amazon**
### Issue #, if available:
https://issues.amazon.com/issues/Chime-41485
### Description of changes:
Just like JS SDK we need an event meetingStartDurationMs - the time that elapsed between the start request audioVideo.start and the beginning of the meeting AudioVideoObserver.audioVideoDidStart.
* Add an attribute `meetingStartDurationMs` in `IngestionPayload`.
* Replace `EventAttributeUtils.getCommonAttributes()` with `EventAttributeUtils.getCommonAttributes(ingestionConfiguration: ingestionConfiguration))` to include `meetingId` and `attendeeId` in metadata, which could enable the metrics.
* Send meetingEnded event immediately when it ends, which could avoid the failure of sending event after the end of meeting.

### Testing done:
Sanity Test done.
#### Manual test cases (add more as needed):

- [ ] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [ ] Leave meeting
- [ ] Rejoin meeting
- [ ] See metrics
- [ ] See attendee presence
- [ ] Send audio
- [ ] Receive audio
- [ ] Mute self
- [ ] Unmute self
- [ ] See local mute indicator when muted
- [ ] See remote mute indicator when other is muted
- [ ] Enable and disable local video
- [ ] Enable and disable remote video from remote side
- [ ] Send local video
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift [hunk 1]
```diff
@@ -11,6 +11,7 @@ import Foundation
 @objcMembers public class DefaultMeetingStatsCollector: NSObject, MeetingStatsCollector {
     private let logger: Logger
     private var meetingStartTimeMs: Int64 = 0
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift:13]: meetingStartTimeMs is the time that meeting is started,
meetingStartConnectingTimeMs is the time that meeting start request is initialized, right?

File: AmazonChimeSDK/AmazonChimeSDK/ingestion/IngestionEventConverter.swift [hunk 2]
```diff
@@ -67,13 +68,13 @@ import Foundation
             return nil
         }
 
-        let rootMeta = toIngestionMetadata(attributes: EventAttributeUtils.getCommonAttributes())
+        let rootMeta = toIngestionMetadata(attributes: EventAttributeUtils.getCommonAttributes(ingestionConfiguration: ingestionConfiguration))
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/ingestion/IngestionEventConverter.swift:71]: Is adding the ingestionConfiguration param related to this event metrics? Have not seen this is used in this PR.

File: AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift (hunk 1 above)
Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift:13]: yes

File: AmazonChimeSDK/AmazonChimeSDK/ingestion/IngestionEventConverter.swift (hunk 2 above)
Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/ingestion/IngestionEventConverter.swift:71]: getCommonAttributes(ingestionConfiguration: ingestionConfiguration) will add meetingId and attendeeId to metadata, which is required by JS team.

**Review by ziyiz-amzn — DISMISSED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift [hunk 3]
```diff
@@ -29,6 +30,8 @@ import Foundation
         return [EventAttributeName.maxVideoTileCount: maxVideoTileCount,
                 EventAttributeName.retryCount: retryCount,
                 EventAttributeName.poorConnectionCount: poorConnectionCount,
+                EventAttributeName.meetingStartDurationMs: meetingStartConnectingTimeMs == 0 ?
```
Comment (hokyungh) [inline AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift:33]: so if meeting failed to start, will this value be negative or should it be 0?

Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift:33]: It will be reset to 0 in resetMeetingStats()

Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift]: updated

**Review by hokyungh — DISMISSED** [review]
cool. You can have CHANGELOG next PR.

**Review by ziyiz-amzn — COMMENTED** [review]
Will need to add CHANGELOG.

**Review by ziyiz-amzn — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]

**Review by ziyiz-amzn — APPROVED** [review]

**Pull request merged**